### PR TITLE
Watch topo relations

### PR DESCRIPTION
**Pull request by rgemawat2000**

### CODE REVIEW

**Comment by rgemawat2000** [comment]
Screen Shot 2020-08-04 at 4 17 06 PM
Here is the duplication in the floatpanel for entity - what should be kept where?
I added the displayname to the title. But in the case that only id exists - id will be shown in displayname and also below it.
I think I can bind it to a boolean with ngIf to only show id below when dislayname exists.

File: web/onos-gui/src/app/onos-topo/topodevice.service.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export class TopoDeviceService {
     topoDevicesSub: Subscription;
     topoEntitySub: Subscription;
     entityList: Map<string, Object>; // Expect <dev-id:dev-ver> as key
+    relationshipsList: Map<string, Object>;
```
Comment (SeanCondon) [inline web/onos-gui/src/app/onos-topo/topodevice.service.ts:39]: perfect

File: web/onos-gui/src/app/onos-topo/topodevice.service.ts [hunk 2]
```diff
@@ -267,27 +269,50 @@ export class TopoDeviceService {
 
     watchTopoEntity(errorCb: (e: grpcWeb.Error) => void, updateCb?: (type: Update.Type, entity: Object) => void) {
         this.topoEntitySub = this.onosTopoDeviceService.requestListTopo().pipe(
-            filter(x => x.getUpdate().getObject().getType() === 1)
+            filter(x => x.getUpdate().getObject().getType() === 1 || x.getUpdate().getObject().getType() === 2)
```
Comment (SeanCondon) [inline web/onos-gui/src/app/onos-topo/topodevice.service.ts:272]: perfect

File: web/onos-gui/src/app/onos-topo/topodevice.service.ts [hunk 3]
```diff
@@ -314,6 +339,23 @@ export class TopoDeviceService {
...
+    removeRelation(name: string): boolean {
+        if (this.relationshipsList.has(name)) {
+            this.relationshipsList.delete(name);
+            return true;
+        }
+        return false;
+    }
+
+    addTopoRelation(obj: Object): boolean {
+        const name = obj.getRelation().getSrcEntityId() + ' ' + obj.getRelation().getTgtEntityId();
```
Comment (SeanCondon) [inline web/onos-gui/src/app/onos-topo/topodevice.service.ts:351]: the relationship itself has an `id` that you should use as the key in the map - i'm not sure it's safe to assume there will not be 2 relationships with the same `src` and `tgt`

File: web/onos-gui/src/app/onos-topo/topodevice.service.ts [hunk 4]
```diff
@@ -267,27 +269,50 @@ export class TopoDeviceService {
...
+                        this.entityList.set(name, updated);
+                    } else {
+                        console.log('Unhandled Topo Entity update', resp.getUpdate().getType(), resp.getUpdate().getObject().getId());
                     }
-                } else if (resp.getUpdate().getType() === Update.Type.MODIFY) {
-                    const updated = resp.getUpdate().getObject();
-                    this.entityList.set(name, updated);
                 } else {
-                    console.log('Unhandled Topo update', resp.getUpdate().getType(), resp.getUpdate().getObject().getId());
+                    const name = resp.getUpdate().getObject().getRelation().getSrcEntityId() + ' '
```
Comment (SeanCondon) [inline web/onos-gui/src/app/onos-topo/topodevice.service.ts:296]: I think name can be the `id` of the relationship object

**Comment by rgemawat2000** [comment]
The displayname error was a really stupid typo on my part - it should be working now.

**Comment by rgemawat2000** [comment]
There are still some issues regarding the duplication - but considering that there are a lot of commits on this patch, I'm not sure whether to push or make another patch.

**Pull request merged**